### PR TITLE
Update CI (#8)

### PR DESCRIPTION
**Pull request by hdholm**
This updates github actions to:

Include current OS versions
Update actions/checkout to version 3 in advance of the imminent NodeJS deprecation
Add dependency installation for CentOS Stream versions
In my initial pass, there were failures for Ubuntu:bionic and, for clang on CentOS 7 and CentOS Stream 9. Rather than spend a lot of time trying to resolve the failures I simply removed the clang attempts for CentOS7 and Stream 9. It's probably some dependency issue if someone wants to chase it. I simply removed Ubuntu:bionic since it's EOL at the end of the month anyway.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -15,12 +15,19 @@ jobs:
         os:
           - fedora:latest
           - quay.io/centos/centos:stream8
-          - quay.io/centos/centos:stream9
```
Comment (sarroutbi) [inline .github/workflows/build.yml:18]: Is this removal necessary?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -34,8 +41,8 @@ jobs:
             os: ubuntu:devel
             stable: false
     steps:
-      - uses: actions/checkout@v2
-
+      - uses: actions/checkout@v3
+
```
Comment (sarroutbi) [inline .github/workflows/build.yml:45]: Remove this empty spaces, please

**Review by sarroutbi — CHANGES_REQUESTED** [review]
Hello @hdholm. Thanks for your PR.

Can you please:
* Clarify why centos:stream9 is removed?
* Remove empty whitespaces?

File: .github/workflows/build.yml (hunk 1 above)
Comment (hdholm) [reply .github/workflows/build.yml:18]: It's in the commit message.  CentOS 7 and Stream 9 got moved out of the first block (which uses both clang and gcc) and into the block below configured simply for gcc.  clang wasn't getting installed correctly on either CentOS 7 nor Stream 9 and although it's probably some rpm/dnf magic I'm overlooking I couldn't figure it out after a little looking so it seemed better to (for the moment at least) just drop the clang compiles to get the CI tests running and hopefully someone can figure out what needs to happen to get clang happy on those two OSs.

File: .github/workflows/build.yml (hunk 2 above)
Comment (hdholm) [reply .github/workflows/build.yml]: Done

File: .github/workflows/build.yml (hunk 1 above)
Comment (sarroutbi) [reply .github/workflows/build.yml:18]: Hello. Thanks for clarifying. I will open an Issue then to track the clang on CentOS Stream 9 issue.

**Review by sarroutbi — APPROVED** [review]
Change LGTM. @sergio-correia: anything to add?

File: .github/workflows/build.yml (hunk 1 above)
Comment (hdholm) [reply .github/workflows/build.yml:18]: I'm not sure why clang in Stream 9 was failing (possibly getting wrong install dependencies section) but it works now with the cleaned up version I just pushed. Only CentOS 7 seems to fail with clang and that's because clang is in the SCL in CentOS 7 and would require changes to the build to run the compiler with the SCL command.  Since it's clearly never worked before and the gcc build works fine and CentOS 7 is approaching EOL, it doesn't seem worth the work to make the changes necessary for clang rather than just live with gcc there.

Comment (sarroutbi) [reply .github/workflows/build.yml:18]: Hello @hdholm. Thanks for the PR. I agree it is not that important to have clang compilation in CentOS7 right now.
In my opinion, this can be merged.

Let's wait a little bit for @sergio-correia to have a view on it and we will merge it before creating the new release.

**Comment by sarroutbi** [comment]
Change LGTM. @sergio-correia : can you please share your view on this change?

**Review by sergio-correia — APPROVED** [review]
Thanks, looks good to me.
